obelix/idefix/services/google/gcalendar.go

```diff
@@ -0,0 +1,125 @@
+package google
+
+import (
+	"encoding/json"
+	"errors"
+	"fmt"
+	"github.com/Shopify/sarama"
+	redisv8 "github.com/go-redis/redis/v8"
+	"google.golang.org/api/calendar/v3"
+	"idefix/operator"
+	"idefix/producer"
+	"idefix/redis"
+	"log"
+	"time"
+)
+
+const (
+	ACTIVE = iota
+	NoActive
+)
+
+type GCalendar struct {
+	date string
+}
+
+func (gc *GCalendar) getEventsList(srv *calendar.Service) (*calendar.Events, error) {
+	t := time.Now().Format(time.RFC3339)
+	events, err := srv.Events.List("primary").
+		ShowDeleted(false).
+		SingleEvents(true).
+		TimeMin(t).
+		MaxResults(1).
+		OrderBy("startTime").
+		Do()
+
+	if err != nil {
+		return nil, fmt.Errorf("failed to retrieve next event gcalendar: %w", err)
+	}
+
+	return events, nil
+}
+
+func (gc *GCalendar) Parse(srv *calendar.Service) error {
+	events, err := gc.getEventsList(srv)
+	if err != nil {
+		return fmt.Errorf("failed to get events list gcalendar: %w", err)
+	}
+
+	if len(events.Items) == 0 {
+		fmt.Println("No upcoming events found.")
+	} else {
+		for _, item := range events.Items {
+			gc.date = item.Start.DateTime
+			if gc.date == "" {
+				gc.date = item.Start.Date
+			}
+		}
+	}
+
+	return nil
+}
+
+func (gc *GCalendar) LookForChange(op *operator.IdefixOperator, key, old string) error {
+	now := time.Now()
+	agendaDate := gc.date
+
+	parse, err := time.Parse(time.RFC3339, agendaDate)
+	if err != nil {
+		log.Fatalln(err)
+	}
+
+	res := parse.Sub(now)
+	if res.Minutes() >= 0 && res.Minutes() <= 10 {
+		if old == string(rune(NoActive)) {
+			err := op.RC.UpdateRedisState(key, string(rune(ACTIVE)))
+			if err != nil {
+				return fmt.Errorf("failed to update redis state: %w", err)
+			}
+
+			err = gc.SendToKafka(op.KP, key)
+			if err != nil {
+				return fmt.Errorf("failed to send message to kafka: %w", err)
+			}
+		} else if old == string(rune(ACTIVE)) {
+			return nil
+		}
```

Comment:
Useless else, you can also do a switch case if you want

Reply:
its much more clear that way